tests/integration/features/partitions/executor/test_collisions.py

```diff
@@ -0,0 +1,167 @@
+            "(default)/file": "(mypart)/file"
+        }
+
+        Path(new_dir / "part1").mkdir()
+        Path(new_dir / "part1/file").write_text("contents")
+        Path(new_dir / "part2").mkdir()
+        Path(new_dir / "part2/file").write_text("contents")
+
+        lcm = LifecycleManager(
+            stub_parts_yaml,
+            application_name="test",
+            cache_dir=Path(),
+            partitions=partitions,
+        )
+
+        with lcm.action_executor() as aex:
+            aex.execute(lcm.plan(Step.STAGE))
+
+        assert Path(new_dir / "stage/file").read_text() == "contents"
+        assert Path(new_dir / "partitions/mypart/stage/file").read_text() == "contents"
+
+    def test_different_file_contents_different_partition(
+        self, new_dir, partitions, stub_parts_yaml
+    ):
+        """Two parts can stage a file with different contents in different partitions."""
+        stub_parts_yaml["parts"]["part1"]["organize"] = {
+            "(default)/file": "(mypart)/file"
+        }
+
+        Path(new_dir / "part1").mkdir()
+        Path(new_dir / "part1/file").write_text("contents")
+        Path(new_dir / "part2").mkdir()
+        Path(new_dir / "part2/file").write_text("conflicting contents")
+
+        lcm = LifecycleManager(
+            stub_parts_yaml,
+            application_name="test",
+            cache_dir=Path(),
+            partitions=partitions,
+        )
+
+        with lcm.action_executor() as aex:
+            aex.execute(lcm.plan(Step.STAGE))
+
+        assert Path(new_dir / "stage/file").read_text() == "conflicting contents"
+        assert Path(new_dir / "partitions/mypart/stage/file").read_text() == "contents"
+
+    def test_different_file_contents_same_partition(
+        self, new_dir, partitions, stub_parts_yaml
+    ):
+        """Two parts cannot stage a file with the different contents in the same partition."""
+        stub_parts_yaml["parts"]["part1"]["organize"] = {
+            "(default)/file": "(mypart)/file"
+        }
+        stub_parts_yaml["parts"]["part2"]["organize"] = {
+            "(default)/file": "(mypart)/file"
+        }
+
+        Path(new_dir / "part1").mkdir()
+        Path(new_dir / "part1/file").write_text("contents")
+        Path(new_dir / "part2").mkdir()
+        Path(new_dir / "part2/file").write_text("conflicting contents")
+
+        lcm = LifecycleManager(
+            stub_parts_yaml,
+            application_name="test",
+            cache_dir=Path(),
+            partitions=partitions,
+        )
+
+        with pytest.raises(PartFilesConflict) as raised:
+            with lcm.action_executor() as aex:
+                aex.execute(lcm.plan(Step.STAGE))
+
+        assert str(raised.value) == (
+            "Failed to stage: parts list the same file with different contents or permissions.\n"
+            "Parts 'part2' and 'part1' list the following files, but with different contents or permissions:\n"
+            "    file"
```

Comment:
Can we have this provide the partition name (when enabled) too? It could speed up debugging for more complex projects to have both a set of parts and a destination partition to look into, especially if the user scripts the partition assignments.

Reply:
Great idea!